tools/generators/legacy/src/DTO/BazelLabel.swift

```diff
@@ -108,6 +108,17 @@ extension BazelLabel: Decodable {
     }
 }
 
+extension BazelLabel {
+    /// The string representation of the label, favoring the shorthand form.
+    ///
+    /// E.g. for `@repo//foo/bar/wiz:wiz` it returns `@repo//foo/bar/wiz`,
+    /// while for `@repo//foo/bar/wiz:baz` it returns `@repo//foo/bar/wiz:baz`.
+    var shortDescription: String {
```

Comment:
By doing this only here, sorting can end up a little weird. I think we should modify the struct to include a bool (calculated once at parsing) that says if it should use short form or not. Then in `description` we can do this. That way sorting works based on the same form that is shown to users.

Reply:
Thanks Brentley!

I moved the logic to `description`, but did not add a bool calculated during parsing.
Couldn't see that being needed, and instead `canOmitName` is calculated inline.
Wouldn't that be ok? Perhaps I misunderstood and you had something else in mind.

Also added and updated some unit tests for `BazelLabel`,
at the same time fixing handling of cases where `name` contains `/` to adhere to what's specified in [the docs](https://bazel.build/concepts/labels#target-names).
(though I think `rules_swift` doesn't handle targets named this way anyway)

Reply:
`description` is called lots of times, in particular because it's called in sorting. Because of this, I would prefer it to be calculated in `init` once. And actually, because it's called so much, let's go a step further and just pre-calculate the description and store it (instead of `canOmitName`).

Reply:
I see, thank you for clarifying!
Pushed that change.